drive/driveimpl/remote_impl.go

```diff
@@ -333,11 +333,15 @@ func (s *userServer) run() error {
 		args = append(args, s.Name, s.Path)
 	}
 	var cmd *exec.Cmd
-	if s.canSudo() {
+	if su := s.canSU(); su != "" {
 		s.logf("starting taildrive file server as user %q", s.username)
-		allArgs := []string{"-n", "-u", s.username, s.executable}
-		allArgs = append(allArgs, args...)
-		cmd = exec.Command("sudo", allArgs...)
+		// Quote and escape arguments. Use single quotes to prevent shell substitutions.
+		for i, arg := range args {
+			args[i] = "'" + strings.ReplaceAll(arg, "'", "'\"'\"'") + "'"
+		}
+		cmdString := fmt.Sprintf("%s %s", s.executable, strings.Join(args, " "))
```

Comment:
Should this bit instead be escaping the arguments, and then quote wrapping the entire `cmdString`? 

e.g., so that we end up with something like `su my-user -c 'example-executable arg1 arg2'`

Reply:
Yes, yes it should.

Reply:
This makes me realize that I have a bigger problem - `su` on MacOS doesn't support the `-c` flag ...  So I think we need to still support sudo on those platforms that don't have a suitable su.

Reply:
> Yes, yes it should.

Take that back. We're passing `cmdString` as an argument to `exec.Command()`, which doesn't require us to quote arguments (remember, we're not in a shell). But, we do still need to quote the actual arguments to avoid command substitution. For example:

```
ubuntu@ip-172-31-4-174:~$ sudo su - -c 'echo $(pwd)'
/root
ubuntu@ip-172-31-4-174:~$ sudo su - -c $'echo \'$(pwd)\''
$(pwd)
```

Reply:
Okay, I updated the PR to still include sudo for when that's available and only treat su as a fallback.

Reply:
> I updated the PR to still include sudo for when that's available and only treat su as a fallback.

Never mind, `-c` is supported on MacOS, so we can just use `su` across the board.